File: 1827-minimum-operations-to-make-the-array-increasing/1827-minimum-operations-to-make-the-array-increasing.cpp
class Solution {
public:
    int minOperations(vector<int>& nums) {
        int n = nums.size();
        if (n == 1) return 0;
        int prev = 0, result = 0;
        for (auto &num : nums) {
            if (num <= prev) {
                result += (prev-num+1);
                prev += 1;
            } else {
                prev = num;
            }
        }
        return result;
    }
};

File: 1827-minimum-operations-to-make-the-array-increasing/1827-minimum-operations-to-make-the-array-increasing.py
class Solution:
    def minOperations(self, nums: List[int]) -> int:
        prev, n, result = 0, len(nums), 0
        if n == 1:
            return 0
        for num in nums:
            if num > prev:
                prev = num
            else:
                result += prev - num + 1
                num = prev + 1
                prev = num
        return result